### PR TITLE
add middleware to handle redirected urls

### PR DESCRIPTION
**Pull request by asmega**

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -23,5 +26,9 @@ class Application < Rails::Application
     config.autoload_paths += %W(#{config.root}/lib)
 
     config.assets.precompile += %w(comfortable_mexican_sofa/admin/basic.css rsvp/rsvp.js)
+
+    if Feature.active?(:redirects)
+      config.middleware.use Rack::RedirectMiddleware::Responder
+    end
```
Comment (tomas-stefano) [inline config/application.rb]: Is this being served through the scripts?

**Comment by tomas-stefano** [comment]
:+1: :+1:  (The minor question is not related to the approval)

**Comment by darokel** [comment]
All seems legit :+1: Good work

File: config/application.rb (hunk 1 above)
Comment (asmega) [inline config/application.rb]: the features toggles are served in `config/features/ENV.yml`, so from the checked in code base and not scripts

Comment (asmega) [inline config/application.rb]: see this PR https://github.com/moneyadviceservice/cms/pull/249

File: lib/rack/redirect_middleware.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    class Responder
+      def initialize(app)
+        @app = app
+      end
+
+      def call(env)
+        regex = /\A\/api(\/[^.]*)(.json)?\z/
+        source_path = Array(regex.match(env['PATH_INFO']))[1]
+
+        if source_path && redirect = Redirect.find_by(source: source_path)
```
Comment (tomas-stefano) [inline lib/rack/redirect_middleware.rb]: :+1:

**Comment by tomas-stefano** [comment]
LGTM :+1: :ship:

**Pull request merged**